### DISCUSSION TITLE
How to fetch the ChangeSet using the changeset id.

### DISCUSSION THREAD

**Discussion by pahm3y** [iTwin/itwinjs-core#4945] [General]
Using the imodeljs library version **2.19.2**.
Using the following API to get the ChangeSet using the changeset id.

/** Get the [[ChangeSet]]s for the iModel.
     * @param requestContext The client request context
     * @param iModelId Id of the iModel. See [[HubIModel]].
     * @param query Optional query object to filter the queried ChangeSets or select different data from them.
     * @returns ChangeSets that match the query.
     * @throws [WsgError]($itwin-client) with [WSStatus.InstanceNotFound]($bentley) if [[InstanceIdQuery.byId]] is used and a [[ChangeSet]] with the specified id could not be found.
     * @throws [Common iModelHub errors]($docs/learning/iModelHub/CommonErrors)
     */
    **get(requestContext: AuthorizedClientRequestContext, iModelId: GuidString, query?: ChangeSetQuery): Promise<ChangeSet[]>;**

For the last parameter in the above API, We are using  
**const queryChange: ChangeSetQuery = new ChangeSetQuery();
queryChange.fromId("<some changeset id value>")**

The above API fails to get the changeset detail giving the following error:
**_Error   |itwin-client.Clients| 400 Error: No parser is registered for the query option._**

Need help in identifying the required API or resolve this issue.
If there are any other API that can give the ChangeSet information if we have the changeset id, please let me know.

**Note: I am bound to use only 2.19.2 version of iModelJs.**

**Comment by pratikshanghosh** [comment]
Please find the below example in which you fetch the changeset details using a change set id :

API :  
[Get iModel Changeset]

GET **https://api.bentley.com/imodels/{id}/changesets/{changeset}**

Request Example :
**https://api.bentley.com/imodels/b85cb594-22fe-48fe-bb55-dad80167740c/changesets/a5141e511894f125dff14fa9a0b084b52bcac189**

**Note: It is recommended to use the version 3.x of iModelJs.**

**Please visit the link for more details : https://developer.bentley.com/apis/imodels/operations/get-imodel-changeset-details/#get-imodel-changeset-details**

**Reply by pahm3y** [reply]
@pratikshanghosh  Thanks for the input. 
I am looking for similar stuff using the backend API not with the https request.

**Comment by pratikshanghosh** [comment]
@pahm3y Can you please explain me the requirement for backend API without http request.
If you can give me an example then it will be helpful for me to understand better.
